spec/models/series_spec.rb

```diff
@@ -110,4 +127,92 @@
       end
     end
   end
+
+  describe "#fandoms" do
+    let(:restricted_work) { create(:work, restricted: true, fandom_string: "Testing2") }
+    let(:hidden_work) { create(:work, hidden_by_admin: true, fandom_string: "Testing3") }
+
+    before do
+      series.works = [unrestricted_work, restricted_work, hidden_work]
+      series.reload
+    end
+
+    shared_examples "only returns fandoms on unrestricted, unhidden works" do
+      it "returns fandoms on unrestricted, unhidden works" do
+        expect(series.fandoms).to include(*unrestricted_work.fandoms)
+      end
+
+      it "does not return fandoms on restricted works" do
+        expect(series.fandoms).not_to include(*restricted_work.fandoms)
+      end
+
+      it "does not return fandoms on hidden works" do
+        expect(series.fandoms).not_to include(*hidden_work.fandoms)
+      end
+    end
+
+    context "when no user is logged in" do
+      it_behaves_like "only returns fandoms on unrestricted, unhidden works"
+    end
+
+    context "when logged in as a regular user" do
+      before do
+        User.current_user = create(:user)
+      end
+
+      it_behaves_like "only returns fandoms on unrestricted, unhidden works"
+    end
+
+    context "when loggged in as an admin" do
+      before do
+        User.current_user = create(:admin)
+      end
+
+      it_behaves_like "only returns fandoms on unrestricted, unhidden works"
+    end
+  end
+
+  describe "#tag_groups" do
+    let(:restricted_work) { create(:work, restricted: true, fandom_string: "Testing2") }
+    let(:hidden_work) { create(:work, hidden_by_admin: true, fandom_string: "Testing3") }
```

Comment:
Could you test this with a different tag type, since Fandom is already mostly covered by the `#fandoms` test?